### PR TITLE
Solves GH-21 add key manager module

### PR DESCRIPTION
**Pull request by sennui**
The key manager doesn't overwrite existing keys
and returns error

### CODE REVIEW

**Comment by sennui** [comment]
Why the unit test fails? Can I mock file or crypto?

File: apps/aecore/src/aec_keys.erl [hunk 1]
```diff
@@ -1,16 +1,290 @@
...
+    Curve = secp256k1,
+
+    %% INFO: set the password to re-use keys between restarts
+    Password = application:get_env(aecore, password, undefined),
+    KeysDir = application:get_env(aecore, keys_dir, filename:join(code:root_dir(), "keys")),
+
+    %% Ensure there is directory for keys
+    case filelib:is_dir(KeysDir) of
+        false ->
+            file:make_dir(KeysDir);
```
Comment (zp-sd) [inline apps/aecore/src/aec_keys.erl]: maybe `ok = file:make_fir(KeysDir)` just in case?

File: apps/aecore/src/aec_keys.erl [hunk 2]
```diff
@@ -1,16 +1,290 @@
...
+            {reply, {error, keys_not_loaded}, State}
+    end.
+
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+handle_info(timeout, #state{priv=undefined, pub=undefined, pass=Password,
+                           type=KeyType, curve=Curve, keys_dir=KeysDir} = State) ->
+    try
+        {PubFile, PrivFile} = p_gen_filename(KeysDir),
```
Comment (zp-sd) [inline apps/aecore/src/aec_keys.erl]: what does `p_` prefix mean?

File: apps/aecore/src/aec_keys.erl [hunk 3]
```diff
@@ -1,16 +1,290 @@
...
+    ok = to_local_dir(PubFilename, EncPub),
+    ok = to_local_dir(PrivFilename, EncPriv),
+    {NewPubKey, NewPrivKey}.
+
+to_local_dir(NewFile, Bin) ->
+    lager:debug("Saving keys to ~p", [NewFile]),
+    case file:read_file(NewFile) of
+        {error, enoent} ->
+            {ok, IODevice} = file:open(NewFile, [write, binary, raw]),
+            file:write_file(NewFile, Bin),
```
Comment (zp-sd) [inline apps/aecore/src/aec_keys.erl]: `ok = (...)`

File: apps/aecore/src/aec_keys.erl [hunk 4]
```diff
@@ -1,16 +1,290 @@
...
+            file:close(IODevice);
```
Comment (zp-sd) [inline apps/aecore/src/aec_keys.erl]: `ok = (...)`

File: apps/aecore/test/aec_keys_tests.erl [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                    107,188,126,242,98,36,211,79,105,50,16,124,227,93,228,142,83,163,126,167,206>>).
+
+-define(TEST_PRIV, <<116,214,52,147,205,225,149,14,95,228,19,253,76,49,82,53,18,39,12,254,53,
+                     220,210,199,152,25,208,77,217,186,141,212>>).
+
+all_test_() ->
+    {setup,
+     fun setup/0,
+     fun teardown/1,
+     [{"Test if eunit is still broken and doesn't pick up sys.config",
```
Comment (zp-sd) [inline apps/aecore/test/aec_keys_tests.erl]: 🤔

File: apps/aecore/test/aec_keys_tests.erl [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+       fun() ->
+               ?assertEqual({error,key_not_found}, aec_keys:pubkey())
+       end}
+     ]
+    }.
```
Comment (zp-sd) [inline apps/aecore/test/aec_keys_tests.erl]: Maybe add also test where you `application:set_env` keys location and fetch keys from disk?

Comment (sennui) [reply apps/aecore/test/aec_keys_tests.erl]: its gambling... how, without profiles, I can know which dirs are available (sure, can try with /tmp...). Seems like a integration test

File: apps/aecore/src/aec_keys.erl (hunk 2 above)
Comment (sennui) [reply apps/aecore/src/aec_keys.erl]: private

**Pull request merged**